servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/Karaf4xServerDelegate.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.server.karaf.core.server;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.fusesource.ide.server.karaf.core.server.subsystems.Karaf4xStartupLaunchConfigurator;
+import org.fusesource.ide.server.karaf.core.util.KarafUtils;
+import org.jboss.ide.eclipse.as.core.server.ILaunchConfigConfigurator;
+
+/**
+ * @author lheinema
+ *
+ */
+public class Karaf4xServerDelegate extends Karaf3xServerDelegate {
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.server.karaf.core.server.KarafServerDelegate#validate()
+	 */
+	@Override
+	public IStatus validate() {
+		// check if the folder exists and the karaf.jar is in place
+		IPath rtLoc = getServer().getRuntime().getLocation();
```

Comment:
NPE in case the runtime doesn't exists anymore on filesystem:
<details>
!ENTRY org.eclipse.jface 4 2 2017-09-11 14:22:13.743
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.jface".
!STACK 0
java.lang.NullPointerException
	at org.fusesource.ide.server.karaf.core.server.Karaf4xServerDelegate.validate(Karaf4xServerDelegate.java:32)
	at org.fusesource.ide.server.karaf.core.server.subsystems.Karaf2xLaunchController.canStart(Karaf2xLaunchController.java:105)
	at org.jboss.ide.eclipse.as.wtp.core.server.behavior.ControllableServerBehavior.canStart(ControllableServerBehavior.java:269)
	at org.eclipse.wst.server.core.internal.Server.canStart(Server.java:1655)
	at org.eclipse.wst.server.ui.internal.view.servers.StartAction.accept(StartAction.java:138)
	at org.eclipse.wst.server.ui.internal.view.servers.AbstractServerAction.selectionChanged(AbstractServerAction.java:85)
	at org.eclipse.wst.server.ui.internal.view.servers.StartAction.selectionChanged(StartAction.java:186)
	at org.eclipse.ui.actions.SelectionProviderAction.selectionChanged(SelectionProviderAction.java:144)
	at org.eclipse.jface.viewers.Viewer$1.run(Viewer.java:155)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.ui.internal.JFaceUtil.lambda$0(JFaceUtil.java:44)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
	at org.eclipse.jface.viewers.Viewer.fireSelectionChanged(Viewer.java:152)
	at org.eclipse.jface.viewers.StructuredViewer.updateSelection(StructuredViewer.java:2180)
	at org.eclipse.jface.viewers.StructuredViewer.handleSelect(StructuredViewer.java:1223)
	at org.eclipse.ui.navigator.CommonViewer.handleSelect(CommonViewer.java:463)
	at org.eclipse.jface.viewers.StructuredViewer$4.widgetSelected(StructuredViewer.java:1252)
	at org.eclipse.jface.util.OpenStrategy.fireSelectionEvent(OpenStrategy.java:240)
	at org.eclipse.jface.util.OpenStrategy.access$4(OpenStrategy.java:235)
	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:400)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4428)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4238)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3817)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1044)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1472)
</details>